external-crates/move/crates/move-compiler/src/parser/lexer.rs

```diff
@@ -335,24 +335,62 @@ impl<'input> Lexer<'input> {
         Ok(text)
     }
 
+    // Trim until reaching whitespace: space, tab, lf(\n) and crlf(\r\n).
+    fn trim_until_whitespace(&self, offset: usize) -> &'input str {
+        let mut text = &self.text[offset..];
+        let mut iter = text.chars();
+        while let Some(c) = iter.next() {
+            if c == ' '
+                || c == '\t'
+                || c == '\n'
+                || (c == '\r' && matches!(iter.next(), Some('\n')))
```

Comment:
This is really needed to clean up after `trim_whitespace_and_comments` fails but we still want to keep going, which is happening only in `advance`, which makes the current structure make sense to me. It is also (to me) fairly  easy to understand because `advance` is the place where error recovery takes place and where can play games with recording diagnostics and proceeding afterwards. If we wanted to incorporate it into `trim_whitespace_and_comments`, I am not even sure what we'd return if it encountered error and advanced to the next whitespace. I am open to suggestions, though, if this feels too ugly...